src/planscape/forsys/cluster_stands.py

```diff
@@ -0,0 +1,261 @@
+                if len(priorities) != len(conditions):
+                    raise Exception(
+                        "expected len(priorities) == len(conditions)")
+                for p in priorities:
+                    if p not in conditions:
+                        raise Exception(
+                            "expected conditions to include priority, %s" %
+                            (p))
+                    v = pixel_dist_to_condition_values[x][y][p]
+                    if v < 0 or v > priority_condition_max_value:
+                        raise Exception("expected condition score to be " +
+                                        "within range, [0, %f]" % (
+                                            priority_condition_max_value))
+
+    def _normalize_condition_values(
+            self,
+            pixel_dist_to_condition_values: dict[int,
+                                                 dict[int, dict[str, float]]],
+        priority_condition_max_value: float
+    ) -> dict[int, dict[int, dict[str, float]]]:
+        normalized_pixel_dist_to_condition_values = {}
+        for x in pixel_dist_to_condition_values.keys():
+            if x not in normalized_pixel_dist_to_condition_values.keys():
+                normalized_pixel_dist_to_condition_values[x] = {}
+            for y in pixel_dist_to_condition_values[x].keys():
+                if y not in normalized_pixel_dist_to_condition_values[x].keys():
+                    normalized_pixel_dist_to_condition_values[x][y] = {}
+                conditions = pixel_dist_to_condition_values[x][y]
+                # Division by priority_condition_max_value forces individual
+                # condition values to be within range, [0, 1].
+                # A subsequent division by sqrt(len(conditions)) forces the l2
+                # norm of the condition vector to be within range, [0, 1].
+                denom = priority_condition_max_value * \
+                    np.sqrt(len(conditions.keys()))
+                for c in conditions.keys():
+                    normalized_pixel_dist_to_condition_values[x][y][c] = \
+                        conditions[c] / denom
+        return normalized_pixel_dist_to_condition_values
+
+    def _normalize_priority_weights(self,
+                                    priority_weights: dict[str, float]
+                                    ) -> dict[str, float]:
+        # The denominator is the l2 norm of the priority weights vector.
+        denom = 0
+        for p in priority_weights.keys():
+            denom = denom + np.square(priority_weights[p])
+        denom = np.sqrt(denom)
+
+        normalized_priority_weights = {}
+        for p in priority_weights.keys():
+            # Division by the l2 norm forces the l2 norm of the priority
+            # weights veector to be 1.
+            normalized_priority_weights[p] = priority_weights[p] / denom
+        return normalized_priority_weights
+
+    def _get_mask(
```

Comment:
nit - i think create_mask would more accurately reflect what the method is doing so init() can be more readable

Reply:
Done.